src/schematools/contrib/django/factories.py

```diff
@@ -131,17 +131,40 @@ def _get_related_name(self):
         If the linked table describes the other end of the relationship,
         this field will also be included in the model.
         """
-        if self.field._parent_table.has_parent_table:
+        parent_table = self.field.table
+        if parent_table.is_nested_table:
             # Won't ever show related name for internal tables
-            return to_snake_case(self.field._parent_table["originalID"])
-        elif self.field._parent_table.is_through_table:
-            # Need this for walking over the through table for the "_links" section.
-            return to_snake_case(f"{self.field._parent_table.id}_through_{self.field.id}")
+            return to_snake_case(parent_table["originalID"])
+        elif parent_table.is_through_table:
+            # This provides a reverse-link from each FK in the M2M table to the linked tables,
+            # allowing to walk *inside* the M2M table instead of over it.
+            # For debugging purposes these names are clarified depending on what direction
+            # the key takes.
+            m2m_field = parent_table.parent_table_field
```

Comment:
Ouch, this took some serious Django relations code-reading.

Reply:
indeed.. ☺️